src/imp_pl.rs

```diff
@@ -9,7 +10,7 @@ use parking_lot::{lock_api::RawMutex as _RawMutex, RawMutex};
 pub(crate) struct OnceCell<T> {
     mutex: Mutex,
     is_initialized: AtomicBool,
-    pub(crate) value: UnsafeCell<Option<T>>,
+    pub(crate) value: UnsafeCell<MaybeUninit<T>>,
```

Comment:
Ah, I've forgot that this is `pub(crate)`, this'll make this refactoring more annoying. Yeah, I guess you were right when you've suggested a separate PR, but we can still go either way. 

To do this cleanly we should first prepare code for refactoring in one commit, and then do refactoring in the second commeit. 

The first phase would be to make `value` here and in std `private`, and instead provide the required accessors (something like `fn as_ptr(&self) -> *mut T`). 

The second phase than should change to `MaybeUninit` only inside this module, without touching anything else outside.